### ISSUE TITLE
joinEager without defaultSelects

### ISSUE THREAD

**Issue by usccbc** [Vincit/objection.js#1573]
I can't get joinEager to work without defining defaultSelects. Is this the expected behavior?  FYI, I am using the knexSnakeCaseMappers.

**Models**

```javascript
class User extends Model {
    static get tableName() {
        return 'users'
    }
}

class Item extends Model {
    static get tableName() {
        return 'items'
    }

    static get relationMappings() {
        return {
            user: {
                relation: Model.BelongsToOneRelation,
                modelClass: User,
                join: {
                    from: 'items.userId',
                    to: 'users.id'
                }
            }
        }
    }
}
```

This works. It returns an array of Items, and each Item has a User that is correctly loaded.

```javascript
Item
.query()
.eager('user')
```

This does not work.  I get back an array of Items, but each associated User is empty. And if you look at the select statement that is generated it does not include the mapped fields (user:user_name, etc)

```javascript
Item
.query()
.joinEager('user')
```

Now, if I define a defaultSelects on User then this works as expected. It generates a correct query with "user.user_name as user:user_name" and all is well.
 
```javascript
Item
.query()
.joinEager({
    user: { $modify: ['defaultSelects'] }
})
```

Is this a bug or am I missing something? I'd rather not have to explicitly define all of the fields within defaultSelects if I don't have to.

Thanks!

**Comment by koskimas** [comment]
Yeah, some feature not working is not intended. You have a something wrong in you setup. Can't know what based on what you are showing here.

**Comment by usccbc** [comment]
Ok thanks @koskimas . I've managed to get a bit closer to the problem. It's has something to do with our AWS hosted Postgres. 

- While creating a full reproduction for you; I set up a local instance of Postgres 11.5, cloned the schema/data, created a stripped down test and it worked perfectly, both eager and joinEager.

- I then switched the connection string to point to our AWS hosted instance of Postgres 11.5 with the exact schema/data and only eager works, joinEager returns {}

**Comment by usccbc** [comment]
I found the problem. It was a knex config issue. When requesting column info, the binding for table_catalog was 'undefined' (see below).  In our knex connection setup for AWS we had only specified {host: , user:, password:} and knex relies on 'database' being defined (knex/.../postgres/query/compiler.js:130). 

All queries were working except for joinEager, since it wasn't able to get the column info. Explains why it was working with explicitly defined defaultSelects.

Added {..., database: 'postgres'} to the knex connection and all is well. 

```
method: 'columnInfo',
  options: {},
  timeout: false,
  cancelOnTimeout: false,
  bindings: [ 'event_participants', undefined ],
  __knexQueryUid: 'bd5bf4f4-36ea-4c80-8945-80c489deb97f',
  sql:
   'select * from information_schema.columns where table_name = ? and table_catalog = ? and table_schema = current_schema()',
  output: [Function: output] }
```
